### PR TITLE
integral action-LQR control for quadcopter UAV 

### PR DESCRIPTION
**Pull request by Astik-2002**
An example attempt to integrate a custom attitude controller for quadcopter UAV. The controller is based on a Linear Quadratic Regulator with integral action to eliminate steady state errors.

### CODE REVIEW

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
need to add to cc_type metadata
we should add explanation of PID and this one in description metadata also

**Comment by Astik-2002** [comment]
> need to add to cc_type metadata we should add explanation of PID and this one in description metadata also

Can you share what needs to be added. Also, what will be the correct way to add additional information and explanation?

**Comment by Hwurzburg** [comment]
not your problem to add PID metadata...thats historical...but line 18 of AC_CustonControl.cpp needs this type added

**Comment by Astik-2002** [comment]
@Hwurzburg kindly review this. Will be commiting hyperparams for making lqr viable for any quadcopter shortly

File: libraries/AC_CustomControl/AC_CustomControl_LQR.cpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
+#include "AC_CustomControl_LQR.h"
+
+#if CUSTOMCONTROL_LQR_ENABLED
+
+#include <GCS_MAVLink/GCS.h>
+
+// table of user settable parameters
```
Comment (Hwurzburg) [inline libraries/AC_CustomControl/AC_CustomControl_LQR.cpp:7]: where are these used?

File: libraries/AC_CustomControl/AC_CustomControl.cpp [hunk 2]
```diff
@@ -8,14 +8,15 @@
...
 // #include "AC_CustomControl_Empty.h"
 #include "AC_CustomControl_PID.h"
+#include "AC_CustomControl_LQR.h"
 #include <GCS_MAVLink/GCS.h>
 
 // table of user settable parameters
 const AP_Param::GroupInfo AC_CustomControl::var_info[] = {
     // @Param: _TYPE
     // @DisplayName: Custom control type
     // @Description: Custom control type to be used
```
Comment (Hwurzburg) [inline libraries/AC_CustomControl/AC_CustomControl.cpp:18]:
```suggestion
    // @Description: Custom control type to be used: See README.md file
```
you need to add a section in the README file for LQR...maybe with a wikipedia link to it also

File: libraries/AC_CustomControl/AC_CustomControl_LQR.cpp (hunk 1 above)
Comment (Astik-2002) [reply libraries/AC_CustomControl/AC_CustomControl_LQR.cpp:7]: These parameters are not currently used. the lqr gains for each axis are hardcoded, and are currently set for a symmetrical 10 inch quadcopter. The details on how to calculate these params for different systems will be included in readme

File: libraries/AC_CustomControl/AC_CustomControl.cpp (hunk 2 above)
Comment (Astik-2002) [reply libraries/AC_CustomControl/AC_CustomControl.cpp:18]: Hello @Hwurzburg, readme changes have been pushed. Kindly review them

**Pull request closed without merging**